src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/arraytrailingcomma/InputArrayTrailingComma.java

```diff
@@ -20,7 +20,7 @@
     {
         1,
         2,
-        3 // violation
+        3 // violation Array should contain trailing comma.
```

Comment:
dream is close to completion:
make it like `// violation 'Array should contain trailing comma.'` to let us has some structure in comments

Reply:
This will require modification of our regexp.

Let's also discuss what to do if we have multiple violations on same line.
Proposal:
```// violation 'violation message 1', 'violation message 2', 'violation message 3'```
Just one con is that it'll result in very long lines in some cases.

Reply:
We will do our best to make all lines to have single violation.

If not possible, such edge cases will use old style verification of expected violations in addition to trailing comments validation

Reply:
Keep going with updates on simple cases and move all non simple to separate issues, to deal with them later on . So at least we will see all cases and it will be easier to make decisions on design

Reply:
> make it like // violation 'Array should contain trailing comma.' to let us has some structure in comments

done